sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,367 @@
+     * @throws NullPointerException if {@link #endpoint(String) endpoint} or
+     * {@link #subscriptionKey(String) subscriptionKey} has not been set.
+     * @throws IllegalArgumentException if {@link #endpoint(String) endpoint} cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsAsyncClient buildAsyncClient() {
+        // Global Env configuration store
+        final Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration().clone() : configuration;
+        // Service Version
+        final TextAnalyticsServiceVersion serviceVersion =
+            version != null ? version : TextAnalyticsServiceVersion.getLatest();
+
+        // Endpoint cannot be null, which is required in request authentication
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+
+        HttpPipeline pipeline = httpPipeline;
+        // Create a default Pipeline if it is not given
+        if (pipeline == null) {
+            // Closest to API goes first, closest to wire goes last.
+            final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+            // Authentications
+            if (tokenCredential != null) {
+                // User token based policy
+                policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+            } else if (subscriptionKey != null) {
+                headers.put(OCP_APIM_SUBSCRIPTION_KEY, subscriptionKey);
+            } else {
+                // Throw exception that credential and tokenCredential cannot be null
+                logger.logExceptionAsError(
+                    new IllegalArgumentException("Missing credential information while building a client."));
+            }
+
+            policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+                buildConfiguration));
+            policies.add(new RequestIdPolicy());
+            policies.add(new AddHeadersPolicy(headers));
+            policies.add(new AddDatePolicy());
+
+            HttpPolicyProviders.addBeforeRetryPolicies(policies);
+            policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+            policies.addAll(this.policies);
+            HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+            pipeline = new HttpPipelineBuilder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+        }
+
+        final TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new TextAnalyticsAsyncClient(textAnalyticsAPI, serviceVersion, clientOptions);
+    }
+
+    /**
+     * Set the default client option for one client.
```

Comment:
This `clientOptions` will be used for every client construction until changed.

Reply:
Yes. If the user want to not using default client options they have to set it to null.